client/test/build.gradle

```
@@ -43,6 +43,8 @@ dependencies {
   api "com.carrotsearch.randomizedtesting:randomizedtesting-runner:${versions.randomizedrunner}"
   api "junit:junit:${versions.junit}"
   api "org.hamcrest:hamcrest:${versions.hamcrest}"
+  api "org.bouncycastle:bc-fips:${versions.bouncycastle_jce}"
```

We could make `bc-fips` runtime only now, right?

```suggestion
  runtimeOnly "org.bouncycastle:bc-fips:${versions.bouncycastle_jce}"
```

---

Its bc of `RestClientTestCase` using `CryptoServicesRegistrar` to checking for FIPS mode. I can make the dependency conditional or invoke `CryptoServicesRegistrar.isInApprovedOnlyMode()` per reflection and let it fail if not on classpath. What do you prefer?

---

Could we keep it system property driven as before (as FipsBuildParams foes)? That would eliminate the  classpath dependencies